### DISCUSSION TITLE
Where can I find the source code for the example on https://xtermjs.org?

### DISCUSSION THREAD

**Discussion by AuroraTea** [sourcelair/xterm.js#5032] [Q&A] [answered]
It happens to meet my needs exactly. I don't want to implement pty, but I need to implement one that sends a whole sentence when the return key is pressed.

**Comment by jerch** [comment, marked as answer]
The website source is located here: https://github.com/xtermjs/xtermjs.org
and the terminal js stuff in particular here: https://github.com/xtermjs/xtermjs.org/blob/master/js/demo.js

But allow me to place an upfront warning here - if you dont connect the terminal to an actual PTY, things will get quite hairy and low-levellish, down to writing single byte inspection for line editing and such in your handlers. Things we take for granted on terminals - you have to implement it yourself.

**Reply by AuroraTea** [reply]
Thank you, I am also using xterm.js to implement the front-end of SSH, this approach allows me to avoid maintaining two separate implementations, even though it is more complex than just one. 😁
